test/unit/bindingpatterns.spec.ts

```diff
@@ -1,92 +1,157 @@
 import * as util from "../util";
 
 const testCases = [
-    { bindingString: "{x}", objectString: "{x: true}", returnVariable: "x" },
-    { bindingString: "[x, y]", objectString: "[false, true]", returnVariable: "y" },
-    { bindingString: "{x: [y, z]}", objectString: "{x: [false, true]}", returnVariable: "z" },
-    { bindingString: "{x: [, z]}", objectString: "{x: [false, true]}", returnVariable: "z" },
-    { bindingString: "{x: [{y}]}", objectString: "{x: [{y: true}]}", returnVariable: "y" },
-    { bindingString: "[[y, z]]", objectString: "[[false, true]]", returnVariable: "z" },
-    { bindingString: "{x, y}", objectString: "{x: false, y: true}", returnVariable: "y" },
-    { bindingString: "{x: foo, y}", objectString: "{x: true, y: false}", returnVariable: "foo" },
+    { bindingString: "{x}", objectString: "{x: true}", returnVariable: "x", expectedResult: true },
+    {
+        bindingString: "[x, y]",
```

Comment:
I don't think it's a good idea to adjust all existing tests for this. It is probably tidier if you just add a new `test.each([<...>])("Binding patterns handle false correctly (%p)"` with your false test cases.